src/sql/src/kafka_util.rs

```diff
@@ -254,51 +315,41 @@ pub async fn create_consumer(
 }
 
 /// Returns start offsets for the partitions of `topic` and the provided
-/// `kafka_time_offset` option.
+/// `START TIME OFFSET` option.
 ///
 /// For each partition, the returned offset is the earliest offset whose
 /// timestamp is greater than or equal to the given timestamp for the
 /// partition. If no such message exists (or the Kafka broker is before
 /// 0.10.0), the current end offset is returned for the partition.
 ///
-/// The provided `kafka_time_offset` option must be a non-zero number:
+/// The provided `START TIME OFFSET` option must be a non-zero number:
 /// * Non-Negative numbers will used as is (e.g. `1622659034343`)
 /// * Negative numbers will be translated to a timestamp in millis
 ///   before now (e.g. `-10` means 10 millis ago)
 ///
-/// If `kafka_time_offset` has not been configured, an empty Option is
+/// If `START TIME OFFSET` has not been configured, an empty Option is
 /// returned.
 pub async fn lookup_start_offsets(
     consumer: Arc<BaseConsumer<KafkaErrCheckContext>>,
     topic: &str,
-    with_options: &BTreeMap<String, SqlValueOrSecret>,
+    offsets: KafkaStartOffsetType,
     now: u64,
 ) -> Result<Option<Vec<i64>>, PlanError> {
-    let time_offset = match with_options.get("kafka_time_offset").cloned() {
-        None => return Ok(None),
-        Some(_) if with_options.contains_key("start_offset") => {
-            sql_bail!("`start_offset` and `kafka_time_offset` cannot be set at the same time.")
-        }
-        Some(offset) => offset,
+    let time_offset = match offsets {
+        KafkaStartOffsetType::StartTimeOffset(time) => time,
```

Comment:
Should `KafkaStartOffsetType::StartOffset` panic here because we need to have already converted it to `StartTimeOffset`?

Reply:
Its the other way around, and now, the user can specify `StartOffset` with `START OFFSET` so its fine for this to just pass through. We could match higher up in `ddl.rs` but I don't think its worth it